### PR TITLE
[RectorGenerator] Test interactive generator

### PR DESCRIPTION
**Pull request by TomasVotruba**
Follow up to https://github.com/rectorphp/rector/pull/4931

### CODE REVIEW

**Comment by TomasVotruba** [comment]
@simivar It required big of `CommandTester` hacking (unfortunatelly its closed design does not allow better DI way), but it works now :)

File: packages/rector-generator/tests/ValueObjectFactory/RectorRecipeInteractiveProviderTest.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        $rectorRecipe = $this->rectorRecipeInteractiveProvider->create();
+
+        // generate
+        $this->rectorRecipeGenerator->generate($rectorRecipe, self::DESTINATION_DIRECTORY);
+
+        // compare it
+        $this->assertDirectoryEquals(__DIR__ . '/Fixture/expected_interactive', self::DESTINATION_DIRECTORY);
+
+        // clear it
+        $this->smartFileSystem->remove(self::DESTINATION_DIRECTORY);
```
Comment (TomasVotruba) [inline packages/rector-generator/tests/ValueObjectFactory/RectorRecipeInteractiveProviderTest.php:85]: @simivar Btw, this is what I meant by comparing generated output with expected generated rule

**Comment by simivar** [comment]
Thank you @TomasVotruba, sorry for not providing it!

**Comment by TomasVotruba** [comment]
No problem. Thank you for adding the interactive mode :+1:  I look forward for new rules

**Pull request merged**